overlord/ifacestate/handlers.go

```diff
@@ -538,8 +558,26 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	chg := task.Change()
+	// content iface has special handling for the default-provider
+	// auto-install
+	defaultProviders := m.defaultContentProviders(snapName)
+	for _, chg := range st.Changes() {
+		if chg.Status().Ready() {
+			continue
+		}
+		for _, t := range chg.Tasks() {
+			if t.Status().Ready() || t.Kind() != "link-snap" {
```

Comment:
because at least core does setup-profiles twice, we probably want to check also for "setup-profiles" here

Reply:
Thanks, I added this!